sqlglot/parser.py

```diff
@@ -4183,9 +4188,11 @@ def _parse_type(self, parse_interval: bool = True) -> t.Optional[exp.Expression]
                 if parser:
                     return parser(self, this, data_type)
                 return self.expression(exp.Cast, this=this, to=data_type)
+
             if not data_type.expressions:
                 self._retreat(index)
-                return self._parse_column()
+                return self._parse_id_var() if fallback_to_identifier else self._parse_column()
```

Comment:
I changed this because a databricks test case was failing:

```sql
CREATE TABLE t (c STRUCT<interval: DOUBLE COMMENT 'aaa'>)
```

We'd parse `interval: DOUBLE` into a `JSONExtract` expression because of the colon struct def. syntax.